nio/block/base.py

```diff
@@ -112,6 +112,83 @@ def stop(self):
         """
         pass  # pragma: no cover
 
+    def set_status(self, status, message="", replace_existing=False):
+        """ Set this block's status and notify the management channel.
+
+        This is a method that a block developer can use to set their block's
+        status. A block in warning or error status may be noted in the
+        system designer or in monitoring tools.
+
+        In general a block in warning status has a problematic condition but
+        may be able to recover on its own (e.g., retrying a connection). A
+        block in error status is in a state that will require a human to
+        resolve the error; either by fixing the error condition or restarting
+        the service.
+
+        Args:
+            status (str or RunnerStatus): A nio.util.runner.RunnerStatus to
+                set the block's statut to. Optionally takes a string if the
+                string is one of (error, warning, ok). Specifying 'error' or
+                'warning' will set those status flags on the block. Specifying
+                any non-error status as a status will clear both any error or
+                warning status flags on the block.
+            message (str): An optional string message to include in the
+                notification of the status change.
+            replace_existing (bool): Whether or not to replace the existing
+                block status when setting the new status. If False, the new
+                status will be added to the block if it is a warning or error
+                status. Any non-error/warning status will clear any error or
+                warning status that was on the block before.
+
+        Returns:
+            RunnerStatus: The current block's status after setting.
+
+        Raises:
+            ValueError: If a string other than "ok", "warning", or "error" is
+                provided as the status
+            TypeError: If an invalid status type or message type is specified
+        """
+        if isinstance(status, str):
+            if status.lower() == 'ok':
+                # An ok status means we want the first non-error status that
+                # is on the block right now.
+                # Default to a created status if the block has no statues left
+                status = RunnerStatus.created
+                for flag, flag_set in self.status.flags.items():
```

Comment:
This was kind of tricky to do, I had to find the first status on the block since `blk.status` isn't actually a `RunnerStatus` but rather a `FlagsEnum` of `RunnerStatus` objects. Is there a better way to do this than what I did here?

Reply:
it is tricky, I think when 'ok' comes in, it should first clear warning or error from self.status, as it is now a guy that likes replacing could do something like:
        blk.set_status('warning', replace_existing=True)
        blk.set_status('ok', replace_existing=True)
and status would remain in `warning`

Reply:
That would maybe explain the jenkins failure the one time too